test/Platforms/Echo.Platforms.AsmResolver.Tests/Emulation/Dispatch/ObjectModel/CastHandlerTest.cs

```diff
@@ -179,5 +181,77 @@ public void CastMatchedRefTypeShouldPushSameObject(CilCode code, Type t1, Type t
             var value = Assert.Single(stack);
             Assert.Equal(address, value.Contents.AsSpan().ReadNativeInteger(Context.Machine.Is32Bit));
         }
+
+        [Fact]
+        public void CastGenericTypeReturnsInt16() {
+            var vm = Context.Machine;
+
+            var genericMethod = ModuleFixture.MockModule.TopLevelTypes
+                .First(t => t.Name == nameof(SimpleClass))
+                .Methods
+                .First(m => m.Name == nameof(SimpleClass.GenericMethod));
+
+            var methodSpecification = new MethodSpecification(genericMethod,
+                new GenericInstanceMethodSignature(ModuleFixture.MockModule.CorLibTypeFactory.Int16));
+
+            short value = 32000;
+
+            var returnValue = vm.Call(methodSpecification, new object[] { value });
+
+            Assert.NotNull(returnValue);
+
+            var result = vm.ObjectMarshaller.ToObject<short>(returnValue!);
```

Comment:
We probably do not want to use the marshaller in these test cases, as it implicitly also tests the marshaller, which is testing a lot of other things implicitly. If there's a bug in the marshaller, this test will now fail too. I think we should replace it with `returnValue.AsSpan().I16` (and ofc the equivalent for the other test cases).